libraries/3rdparty/tracealyzer_recorder/Include/trcPortDefines.h

```diff
@@ -70,6 +70,23 @@
 #define TRC_RECORDER_BUFFER_ALLOCATION_DYNAMIC  (0x01)
 #define TRC_RECORDER_BUFFER_ALLOCATION_CUSTOM   (0x02)
 
+/* Default buffer size */
+#ifndef TRC_CFG_NSTREAMBUFFER
```

Comment:
It's not related to unit test cases implementation, but because we are making those  TRC_CFG_NSTREAMBUFFER macros defined only in each boards config folders.

https://github.com/aws/amazon-freertos/blob/master/vendors/pc/boards/windows/aws_tests/config_files/trcSnapshotConfig.h#L132

And Linux as board doesn't have those configuration which makes trcKernelPort.h could not find the reference for TRC_CFG_NSTREAMBUFFER.

https://github.com/aws/amazon-freertos/blob/master/libraries/3rdparty/tracealyzer_recorder/Include/trcKernelPort.h#L508

And causing build errors.

This is a fix I've did to make it compile, but we can do other options like:

1. make it reference https://github.com/aws/amazon-freertos/blob/master/libraries/3rdparty/tracealyzer_recorder/config/trcSnapshotConfig.h

2. make Linux as board also have config folders to add trcSnapshotConfig.h

3. use what we have in current commit.

Feel free to let me know which one is the suggested option.

Reply:
Per offline conversation we are going to take what is needed from: https://github.com/aws/amazon-freertos/pull/2470/files

Reply:
will remove this fix from the PR.

Reply:
fixed in 5c6afe0c590c76d8c90a6972c9798058702963f1

resolving this comment, feel free to let me know if need more work.